var/spack/repos/builtin/packages/py-torch/130443.patch

```diff
@@ -0,0 +1,59 @@
+From 04fc29b4a7709366ab03d15d84d3cf4d7083c82f Mon Sep 17 00:00:00 2001
+From: Harmen Stoppels <me@harmenstoppels.nl>
+Date: Wed, 10 Jul 2024 18:05:00 +0200
+Subject: [PATCH] flash_attn: limit parallelism in build
+
+The flash attention cuda files require an enormous amount of memory and
+on most systems cause OOM errors when compiling with -j $(nproc).
+
+Compiling those files with a ninja pool fixes the problem.
+---
+ aten/src/ATen/CMakeLists.txt | 8 ++++++--
+ caffe2/CMakeLists.txt        | 4 ++++
+ 2 files changed, 10 insertions(+), 2 deletions(-)
+
+diff --git a/aten/src/ATen/CMakeLists.txt b/aten/src/ATen/CMakeLists.txt
+index bf425af5fa..91799773e3 100644
+--- a/aten/src/ATen/CMakeLists.txt
++++ b/aten/src/ATen/CMakeLists.txt
+@@ -177,9 +177,13 @@ file(GLOB mem_eff_attention_cuda_cu "native/transformers/cuda/mem_eff_attention/
+ file(GLOB mem_eff_attention_cuda_kernels_cu "native/transformers/cuda/mem_eff_attention/kernels/*.cu")
+ file(GLOB mem_eff_attention_cuda_cpp "native/transformers/cuda/mem_eff_attention/*.cpp")
+ 
++# flash_attn for cuda requires excessive memory, so restrict parallelism in ninja builds
++add_library(flash_attention_cuda OBJECT)
++set_property(GLOBAL PROPERTY JOB_POOLS flash_attention_pool=3)
```

Comment:
From your PyTorch PR:

> This is done with a ninja job pool. I've hard-coded the number of jobs to 3, since 16GB of memory sounds reasonably standard.

I would disagree with this. A brand new 2024 ~$2K MacBook Pro, a common platform for deep learning, only comes with 8 GB of RAM in the default configuration. I think it would be safer to use a single thread unless we know it would result in a significant slowdown of the overall build. Ideally, your flash-attention issue will get some attention and this patch will be obsolete someday anyway.

Reply:
Okay, then replace darwin with linux. Apple isn't the only company that makes laptops with 8 GB of RAM.